src/components/TextField/TextField.spec.jsx

```diff
@@ -143,4 +143,69 @@ describe('TextField', () => {
 
 		assert(event.persist.called);
 	});
+
+	it('should flush the `onChangeDebounced`-handled event before blurring', () => {
+		const onBlur = sinon.spy();
+		const internalChangeSpy = sinon.spy();
+		const onChangeDebounced = () => {
+			assert(onBlur.notCalled);
+			internalChangeSpy();
```

Comment:
this seems like a hack, so if someone wants to suggest a way to ensure the order of operations in a more clean way, i'm all 👂